core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala

```diff
@@ -76,6 +76,11 @@ class ResourceProfile(
     executorResources.asJava
   }
 
+  /**
+   * Target executor's resource profile id, used for schedule.
+   */
+  def targetExecutorRpId: Int = id
```

Comment:
I don't understand what this is and how is it different then the id?  This needs more explanation.

Reply:
`id` is an identifier for a `ResourceProfile`, the `targetExecutorRpId` describes which executors shall `TaskScheduler` assign tasks. Before this PR, `TaskScheduler` will do exactly id matching between task rp id and executor rp id, so `id` should be enough. 

But if we want to share/reuse executors, `TaskScheduler` will need to schedule tasks to different/compatible executors, so task's RP Id and executor's RP Id may be not same in that way. The `targetExecutorRpId` here try to describe which executors to assign for tasks with `ResourceProfile`, it could be different with `id`.

Pretty like compatible resource profile id in concept.

Reply:
this assume resource profile has to know there is a taskResourceProfile. This also seems to limit you to one executor resource profile that is compatible?   Why can't it match on any number of compatible executors.  I guess with the current implementation you limit it to dynamic allocation off so that is why you can but I don't want to make API changes based on that limit.

Reply:
Don't think we have the actual task RP id and executor's RP id for now but they're all resource profile id still.

Should override `_id` as `_id=DEFAULT_RESOURCE_PROFILE_ID` in `TaskResourceProfile` help get rid of this API?

In this way, the task still limits one executor resource profile. But it doesn't mean dynamic allocation has to be turned off, right? 

If we really want to achieve "match on any number of compatible executors" (which I think is a good direction), I think we need to separate the resource profile id into task resource requirement id and executor resource requirement id completely.

Reply:
> the id if public, so if I create a task resource profile that has the same id as another one, that seems odd right?

Right, right..after second thinking, it's pretty odd 😅, though I think it should be protected not public at least.

Reply:
What if we pass `ResourceProfile.taskResources` to `TaskSet` directly, and pass `ResourceProfile.executorResources` to `WorkerOffer`? It seems like we no longer depend on resource profile id for scheduling in this way.. cc @ivoson @tgravescs

Reply:
Thanks @Ngone51 
I think that's pretty much the idea of sharing/re-use executors with the policy: sharing any executors which can fulfill task's resource requests. The problem may be that do we want to let users to specify re-use policy.

How about we narrow down the scenario here in this PR, schedule tasks with `TaskResourceProfile` to executors with `DEFAULT_RESOURCE_PROFILE` directly when dynamic allocation off(without checking rpId).
Even thought the idea is pretty much like reuse compatible executors, it's much simpler in this case. And we can still leave [SPARK-36699](https://issues.apache.org/jira/browse/SPARK-36699) to handle the API change, we don't introduce the API change in this PR.

What do you think? @tgravescs @Ngone51

Reply:
I'm fine narrowing case here but I don't want to have public api affected. so if you just mean hardcode scheduler or something that might be fine.  We still need to make sure if user passes TaskResourceProfile we error out if not this specific case.

Reply:
Yes, we can limit the use case of `TaskResourceProfile`. I'll make the change and update the PR soon. Thanks.

Reply:
> Even thought the idea is pretty much like reuse compatible executors, it's much simpler in this case. And we can still leave [SPARK-36699](https://issues.apache.org/jira/browse/SPARK-36699) to handle the API change, we don't introduce the API change in this PR.

SGTM! Thanks @ivoson